### DISCUSSION TITLE
How to transfer a max min problem with dualize function

### DISCUSSION THREAD

**Discussion by bmmddzgj** [yalmip/YALMIP#1416] [Optimization modelling] [answered]
I am dealing with two stage robust unit commitment in the following form
![image](https://github.com/yalmip/YALMIP/assets/51513025/34d52700-35d7-4ece-912f-e3401568720e)
![image](https://github.com/yalmip/YALMIP/assets/51513025/df3cdd43-4e90-458c-89d3-1a0580ee7b2d)
According to the well-known column and constraint generation algorithm, such model (6)-(24) can be divided into a master problem (25)-(28) and a subproblem. The subproblem is a max-min problem and can be transformed into a single level problem (29)-(38) by dual theory. Here's the transformed formulation.
![image](https://github.com/yalmip/YALMIP/assets/51513025/50ebd8b8-d880-4b86-8499-a614d3dca7d4)
![image](https://github.com/yalmip/YALMIP/assets/51513025/8779c96f-706f-4d06-8837-9ff7be772896)

I want to code with `dualize` function using yalmip. My problem is if I can apply dualize function to get the above subproblem. I want to keep the uncertainty parameter $u$ as a variable in it and add extra constraints (30)-(31) after getting the initial dual constraints (33)-(38).

By the way, I have tried to write the subproblem but ended in getting an unbounded tips.

**Comment by johanlofberg** [comment, marked as answer]
Most likely you are not supposed to use dualize, it does something very esoteric (it *interprets* a model in primal vs dual space)

**Reply by bmmddzgj** [reply]
Thank you. I still wanna ask if I can leverage robust optimization to handle the second level problem (16)-(24) with given $\mathcal{X}$ by yalmip.

**Reply by johanlofberg** [reply]
To complicated to see in all the details. YALMIP can compute z in {minimize c'*z subject to Az+By<=b for all y in Y} where Y is polytopic, and the "can solve" in practice depends on the complexity of Y and its consequences on deriving the robust counterpart, i.e a problem where y and the for-all operator are eliminated
